scipy/io/tests/test_mmio.py

```diff
@@ -19,97 +19,119 @@ def setUp(self):
-        mmwrite(fn,a)
-        assert_equal(mminfo(fn),(2,2,4,'array','integer','symmetric'))
-        b = mmread(fn)
-        assert_array_almost_equal(a,b)
-
-    def test_simple_skew_symmetric(self):
-        a = [[1,2],[-2,4]]
-        fn = self.fn
-        mmwrite(fn,a)
-        assert_equal(mminfo(fn),(2,2,4,'array','integer','skew-symmetric'))
-        b = mmread(fn)
-        assert_array_almost_equal(a,b)
+    def setUp(self):
+        self.tmpdir = mkdtemp()
+        self.fn = os.path.join(self.tmpdir, 'testfile.mtx')
 
+    def tearDown(self):
+        shutil.rmtree(self.tmpdir)
+    
+    def check(self, a, info):
+        mmwrite(self.fn, a)
+        assert_equal(mminfo(self.fn), info)
+        b = mmread(self.fn)
+        assert_array_almost_equal(a.todense(), b.todense())
+    
+    def test_simple_integer(self):
+        self.check(scipy.sparse.csr_matrix([[1, 2], [3, 4]]), (2, 2, 4, 'coordinate', 'integer', 'general'))
+    
+    def test_simple_upper_triangle_integer(self):
+        self.check(scipy.sparse.csr_matrix([[0, 1], [0, 0]]), (2, 2, 1, 'coordinate', 'integer', 'general'))
+    
+    def test_simple_lower_triangle_integer(self):
+        self.check(scipy.sparse.csr_matrix([[0, 0], [1, 0]]), (2, 2, 1, 'coordinate', 'integer', 'general'))
+    
+    def test_simple_rectangular_integer(self):
+        self.check(scipy.sparse.csr_matrix([[1, 2, 3], [4, 5, 6]]), (2, 3, 6, 'coordinate', 'integer', 'general'))
+    
+    def test_simple_rectangular_float(self):
+        self.check(scipy.sparse.csr_matrix([[1, 2], [3.5, 4], [5, 6]]), (3, 2, 6, 'coordinate', 'real', 'general'))
+    
+    def test_simple_float(self):
+        self.check(scipy.sparse.csr_matrix([[1, 2], [3, 4.0]]), (2, 2, 4, 'coordinate', 'real', 'general'))
+    
+    def test_simple_complex(self):
+        self.check(scipy.sparse.csr_matrix([[1, 2], [3, 4j]]), (2, 2, 4, 'coordinate', 'complex', 'general'))
+    
+    def test_simple_symmetric_integer(self):
+        self.check(scipy.sparse.csr_matrix([[1, 2], [2, 4]]), (2, 2, 3, 'coordinate', 'integer', 'symmetric'))
+    
+    def test_simple_skew_symmetric_integer(self):
+        self.check(scipy.sparse.csr_matrix([[1, 2], [-2, 4]]), (2, 2, 3, 'coordinate', 'integer', 'skew-symmetric'))
+    
     def test_simple_skew_symmetric_float(self):
-        a = array([[1,2],[-2.0,4]],'f')
-        fn = self.fn
-        mmwrite(fn,a)
-        assert_equal(mminfo(fn),(2,2,4,'array','real','skew-symmetric'))
-        b = mmread(fn)
-        assert_array_almost_equal(a,b)
-
-    def test_simple_hermitian(self):
-        a = [[1,2+3j],[2-3j,4]]
-        fn = self.fn
-        mmwrite(fn,a)
-        assert_equal(mminfo(fn),(2,2,4,'array','complex','hermitian'))
-        b = mmread(fn)
-        assert_array_almost_equal(a,b)
-
-    def test_random_symmetric_real(self):
-        sz = (20,20)
+        self.check(scipy.sparse.csr_matrix(array([[1,2], [-2.0,4]], 'f')), (2, 2, 3, 'coordinate', 'real', 'skew-symmetric'))
```

Comment:
@WarrenWeckesser : Thanks for the hint.

I removed (nearly) all PEP8 error which would normally be ignored. This was not fun because there were hundreds and I even did not inserted these.